# Genome-wide Loci Marine Haplosclerids

Joëlle van der Sprong 1§; Nicole J. de Voogd 2,3; Grace P. McCormack 4; Oliver Voigt 1; Dirk Erpenbeck 1,5; Gert Wörheide 1,5,6 ; Sergio Vargas 1§

1 Department of Earth and Environmental Sciences, Palaeontology and Geobiology, Ludwig-Maximilians-Universität München, Munich, Germany

2 Naturalis Biodiversity Center, Leiden, The Netherlands

3 Institute of Environmental Sciences, Universiteit Leiden, Leiden, The Netherlands

4 Molecular Evolution and Systematics Laboratory, MRI, Zoology Department, National University of Ireland Galway, Galway, Ireland

5 GeoBio-Center, Ludwig-Maximilians-Universität München, Munich, Germany

6 Bavarian State Collections of Palaeontology and Geology, Munich, Germany

§ corresponding authors: j.sprong@lmu.de & s.vargas@lmu.de

## Abstract
[insert here ...]

## Description
This repository includes the following:

* The parameters and settings used for bait design (PHYLUCE Tutorial IV) (Faircloth et al. 2012; Faircloth, 2016)
* Transcriptomic data (assemblies) used for bait design (assembled using TransPi; Rivera-Vicéns et al. 2022)
* The MSDA bait set (20,000 probes)
* Phylip-file used for the Maximum Likelihood (ML) full length 18S phylogeny using RAxML (Stamatakis, 2014)
* The parameters and settings for the in-silico test (PHYLUCE Tutorial I) (Faircloth et al. 2012; Faircloth, 2016)
* Assemblies of the Red Sea haplosclerids
* The parameters and settings for reconstructing the ML phylogenies using RAxML (Stamatakis, 2014)
* The parameters and settings for reconstructing the bayesian phylogenies using RevBayes (Höhna et al. 2016)

## Publication

### Citation

## Acknowledgements
Funding was provided by the Deutsche Forschungsgemeinschaft (DFG, project number [...]). We would like to acknowledge that the scientific research cooperation between King Abdulaziz University (KAU, Faculty of Marine Sciences (FMS), Jeddah, Saudi Arabia, and the Senckenberg, Research Institute (SRI), Frankfurt, Germany, in the framework of the Red Sea Biodiversity Project, during which parts of the present material were collected, was funded by KAU GRANT NO. I/1/432-DSR. The authors acknowledge, with thanks, KAU and SRI for technical and financial support. In particular, we would like to thank the Coastal and Marine Resources Core Lab (KAUST) and Mike Berumen for their help in the field. Furthermore, we would like to thank Kenneth Sandoval (National University of Ireland, Galway) and Ignite for providing us with transcriptomic data of haplosclerid sponges. Also, we would like to acknowledge Simone Schätzle, Gabi Büttner, Nora Dotzler, and Vani Tirumalasetty for their help and effort during laboratory work. Finally, we would like to thank Sebastian Hönha and Luiza Guimarães Fabreti for their help and insights into RevBayes and Bayesian inference.

### References
Faircloth, B. C. (2016). PHYLUCE is a software package for the analysis of conserved genomic loci. Bioinformatics, 32(5), 786–788. doi: 10.1093/bioinformatics/btv646

Faircloth, B. C., McCormack, J. E., Crawford, N. G., Harvey, M. G., Brumfield, R. T., & Glenn, T. C. (2012). Ultraconserved elements anchor thousands of genetic markers spanning multiple evolutionary timescales. Systematic Biology, 61(5), 717–726. doi: 10.1093/sysbio/sys004

Höhna, S., Landis, M. J., Heath, T. A., Boussau, B., Lartillot, N., Moore, B. R., … Ronquist, F. (2016). RevBayes: Bayesian phylogenetic inference using graphical models and an interactive model-specification language. Systematic Biology, 65(4), 726–736. doi: 10.1093/sysbio/syw021

Stamatakis, A. (2014). RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics, 30(9), 1312–1313. doi: 10.1093/bioinformatics/btu033

Rivera-Vicéns, R. E., Garcia Escudero, C., Conci, N., Eitel, M., & Wörheide, G. (2022). TransPi – a comprehensive TRanscriptome ANalysiS PIpeline for de novo transcriptome assembly. Molecular Ecology Resources. doi: 10.1111/1755-0998.13593
