# Genome-wide Loci Marine Haplosclerids

Joëlle van der Sprong <sup>1§</sup>; Nicole J. de Voogd <sup>2,3</sup>; Grace P. McCormack <sup>4</sup>; Kenneth Sandoval <sup>4</sup>; Simone Schätzle <sup>1</sup>; Oliver Voigt <sup>1</sup>; Dirk Erpenbeck <sup>1,5</sup>; Gert Wörheide <sup>1,5,6</sup>; Sergio Vargas <sup>1§</sup>

<sup>1</sup> Department of Earth and Environmental Sciences, Palaeontology and Geobiology, Ludwig-Maximilians-Universität München, Munich, Germany

<sup>2</sup> Naturalis Biodiversity Center, Leiden, The Netherlands

<sup>3</sup> Institute of Environmental Sciences, Universiteit Leiden, Leiden, The Netherlands

<sup>4</sup> Molecular Evolution and Systematics Laboratory, MRI, Zoology Department, National University of Ireland Galway, Galway, Ireland

<sup>5</sup> GeoBio-Center, Ludwig-Maximilians-Universität München, Munich, Germany

<sup>6</sup> Bavarian State Collections of Palaeontology and Geology, Munich, Germany

<sup>§</sup> corresponding authors: j.sprong@lmu.de & s.vargas@lmu.de

## Abstract
With declining biodiversity worldwide, a better understanding of species diversity and their relationships is imperative for conservation and management efforts. Marine sponges are species-rich ecological key players on coral reefs, but their species diversity is still poorly understood. This is particularly true for the demosponge order Haplosclerida, whose systematic relationships are contentious due to the incongruencies between morphological and molecular phylogenetic hypotheses. The single gene markers applied in previous studies did not resolve these discrepancies. Hence, there is a high need for a genome-wide approach to derive a phylogenetically robust classification and understand this group's evolutionary relationships. To this end, we developed a target enrichment-based multilocus probe assay for the order Haplosclerida using transcriptomic data. This probe assay consists of 20,000 enrichment probes targeting 2956 ultraconserved elements in coding (i.e. exon) regions across the genome and was tested on 26 haplosclerid specimens from the Red Sea. Our target-enrichment approach correctly placed our samples in a well-supported phylogeny, in agreement with previous haplosclerid molecular phylogenies. Our results demonstrate the applicability of high-resolution genomic methods in a systematically complex marine invertebrate group and provide a promising approach for robust phylogenies of Haplosclerida. Subsequently, this will lead to biologically unambiguous taxonomic revisions, better interpretations of biological and ecological observations and new avenues for applied research, conservation and managing declining marine diversity.

## Description
This repository includes the following:

* The parameters and settings used for bait design (PHYLUCE Tutorial IV)
* Transcriptomic data (assemblies) used for bait design (assembled using TransPi)
* The MSDA bait set (20,000 probes): *uce-20k-probes.fasta*
* Folder with all the information to (re)create the Maximum Likelihood (ML) full length 18S phylogeny using RAxML: *18S_Redmond_phylogeny* 
* The parameters and settings for the in-silico test (PHYLUCE Tutorial I)
* Folder with information and data for in-silico test: *data_insilico_test*
* Folder with information and data for in-vitro test: *data_invitro_test*
* Assemblies of the Red Sea haplosclerids: *assemblies_redsea_haplosclerids*
* The parameters and settings for reconstructing the ML phylogenies using RAxML
* The parameters and settings for reconstructing the bayesian phylogenies using RevBayes

### Citation
van der Sprong, J., de Voogd, N.J., McCormack, G.P., Sandoval, K., Schätzle, S., Voigt, O., Erpenbeck, D., Wörheide, G. and Vargas, S., 2024. A novel target‐enriched multilocus assay for sponges (Porifera): Red Sea Haplosclerida (Demospongiae) as a test case. Molecular Ecology Resources, 24(2), p.e13891.




<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
