# Genome-wide Loci Marine Haplosclerids

Joëlle van der Sprong <sup>1§</sup>; Nicole J. de Voogd <sup>2,3</sup>; Grace P. McCormack <sup>4</sup>; Oliver Voigt <sup>1</sup>; Dirk Erpenbeck <sup>1,5</sup>; Gert Wörheide <sup>1,5,6</sup>; Sergio Vargas <sup>1§</sup>

<sup>1</sup> Department of Earth and Environmental Sciences, Palaeontology and Geobiology, Ludwig-Maximilians-Universität München, Munich, Germany

<sup>2</sup> Naturalis Biodiversity Center, Leiden, The Netherlands

<sup>3</sup> Institute of Environmental Sciences, Universiteit Leiden, Leiden, The Netherlands

<sup>4</sup> Molecular Evolution and Systematics Laboratory, MRI, Zoology Department, National University of Ireland Galway, Galway, Ireland

<sup>5</sup> GeoBio-Center, Ludwig-Maximilians-Universität München, Munich, Germany

<sup>6</sup> Bavarian State Collections of Palaeontology and Geology, Munich, Germany

<sup>§</sup> corresponding authors: j.sprong@lmu.de & s.vargas@lmu.de

## Abstract
[insert here ...]

## Description
This repository includes the following:

* The parameters and settings used for bait design (PHYLUCE Tutorial IV) <sup>a,b</sup>
* Transcriptomic data (assemblies) used for bait design (assembled using TransPi) <sup>c</sup>
* The MSDA bait set (20,000 probes): ####'uce-20k-probes.fasta'
* Folder with all the information to (re)create the Maximum Likelihood (ML) full length 18S phylogeny using RAxML<sup>d,e</sup>: #####'18S_Redmond_phylogeny' 
* The parameters and settings for the in-silico test (PHYLUCE Tutorial I) <sup>a,b</sup>
* Assemblies of the Red Sea haplosclerids: ####'assemblies_redsea_haplosclerids'
* The parameters and settings for reconstructing the ML phylogenies using RAxML <sup>d</sup>
* The parameters and settings for reconstructing the bayesian phylogenies using RevBayes <sup>f</sup>

### Citation
[insert here ...]

### References
<sup>a</sup> Faircloth, B. C. (2016). PHYLUCE is a software package for the analysis of conserved genomic loci. Bioinformatics, 32(5), 786–788. doi: 10.1093/bioinformatics/btv646

<sup>b</sup> Faircloth, B. C., McCormack, J. E., Crawford, N. G., Harvey, M. G., Brumfield, R. T., & Glenn, T. C. (2012). Ultraconserved elements anchor thousands of genetic markers spanning multiple evolutionary timescales. Systematic Biology, 61(5), 717–726. doi: 10.1093/sysbio/sys004

<sup>c</sup> Rivera-Vicéns, R. E., Garcia Escudero, C., Conci, N., Eitel, M., & Wörheide, G. (2022). TransPi – a comprehensive TRanscriptome ANalysiS PIpeline for de novo transcriptome assembly. Molecular Ecology Resources. doi: 10.1111/1755-0998.13593

<sup>d</sup> Stamatakis, A. (2014). RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics, 30(9), 1312–1313. doi: 10.1093/bioinformatics/btu033

<sup>e</sup> Redmond, N. E., Morrow, C. C., Thacker, R. W., Diaz, M. C., Boury-esnault, N., Cárdenas, P., … Collins, A. G. (2013). Phylogeny and systematics of demospongiae in light of new small-subunit ribosomal DNA (18S) sequences. Integrative and Comparative Biology, 53(3), 388–415. doi:10.1093/icb/ict078

<sup>f</sup> Höhna, S., Landis, M. J., Heath, T. A., Boussau, B., Lartillot, N., Moore, B. R., … Ronquist, F. (2016). RevBayes: Bayesian phylogenetic inference using graphical models and an interactive model-specification language. Systematic Biology, 65(4), 726–736. doi: 10.1093/sysbio/syw021




<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
